cli/tests/pcluster/api/controllers/test_cluster_instances_controller.py

```diff
@@ -5,43 +5,167 @@
+                        },
+                    ],
+                    "nextToken": "",
+                },
+            ),
+        ],
+        ids=["empty", "all"],
+    )
+    def test_successful_request(self, mocker, client, instances, params, expected_response):
+        describe_instances_response = []
+        for instance in instances:
+            describe_instances_response.append(cfn_describe_instances_mock_response(**instance))
+        mocker.patch(
+            "pcluster.aws.ec2.Ec2Client.describe_instances",
+            return_value=(describe_instances_response, ""),
+        )
+        response = self._send_test_request(client, **params)
+        with soft_assertions():
+            assert_that(response.status_code).is_equal_to(200)
+            assert_that(response.get_json()).is_equal_to(expected_response)
+
+    @pytest.mark.parametrize(
+        "cluster_name, node_type, queue_name",
+        [
+            ("clustername", None, None),
+            ("clustername", NodeType.HEAD, None),
+            ("clustername", NodeType.COMPUTE, None),
+            ("clustername", None, "queuename"),
+        ],
+        ids=["all instances", "head only", "compute only", "queuename only"],
+    )
+    def test_filters(self, mocker, client, cluster_name, node_type, queue_name):
+        describe_instances_mock = mocker.patch(
+            "pcluster.aws.ec2.Ec2Client.describe_instances",
+            return_value=([cfn_describe_instances_mock_response()], ""),
+        )
+        response = self._send_test_request(
+            client, cluster_name=cluster_name, node_type=node_type, queue_name=queue_name
         )
-        assert_that(response.status_code).is_equal_to(200)
+        with soft_assertions():
+            assert_that(response.status_code).is_equal_to(200)
+            actual_filters = describe_instances_mock.call_args_list[0][0][0]
+            actual_filters_dict = {}
+            for filter in actual_filters:
+                actual_filters_dict[filter["Name"]] = filter["Values"][0]
+            assert_that(actual_filters_dict.get("tag:parallelcluster:cluster-name")).is_equal_to(cluster_name)
+            if node_type:
+                if node_type == NodeType.HEAD:
+                    expected_value = "HeadNode"
+                else:
+                    expected_value = "Compute"
+            else:
+                expected_value = None
+            assert_that(actual_filters_dict.get("tag:parallelcluster:node-type")).is_equal_to(expected_value)
+            assert_that(actual_filters_dict.get("tag:QueueName")).is_equal_to(queue_name)
+
+    @pytest.mark.parametrize(
+        "params, expected_response",
+        [
+            (
+                {"node_type": "wrong_node_type"},
+                {"message": "Bad Request: 'wrong_node_type' is not one of ['HEAD', 'COMPUTE']"},
+            )
+        ],
+    )
+    def test_malformed_request(self, client, mocker, params, expected_response):
```

Comment:
please add a test for bad region and one for bad cluster name (pick a name with invalid chars or < 5 chars)

Reply:
Done